pkg/rancher-desktop/integrations/windowsIntegrationManager.ts

```diff
@@ -403,6 +404,33 @@ export default class WindowsIntegrationManager implements IntegrationManager {
     );
   }
 
+  protected async syncHostFile() {
+    await Promise.all(
+      (await this.supportedDistros).map((distro) => {
+        return this.updateHostFile(distro.name);
+      }),
+    );
+  }
+
+  protected async updateHostFile(distro: string) {
+    if (this.settings.experimental?.virtualMachine?.networkingTunnel) {
+      const vethIPAddr = '192.168.1.2';
+      const contents = await fs.promises.readFile(`\\\\wsl$\\${ distro }\\etc\\hosts`, 'utf-8');
+      const lines = contents.split(/\r?\n/g)
+        .filter(line => !line.includes('host.docker.internal'));
+      const hosts = ['host.rancher-desktop.internal', 'host.docker.internal'];
+      const extra = [
+        '# BEGIN Rancher Desktop configuration.',
```

Comment:
This sounds like a job for [`manageLinesInFile`](https://github.com/rancher-sandbox/rancher-desktop/blob/5161e06b7bf209f23e951e905b41512e9e93c54c/pkg/rancher-desktop/integrations/manageLinesInFile.ts).